volume/modules/local/quota.go

```diff
@@ -0,0 +1,139 @@
+package local
+
+import (
+	"regexp"
+	"strconv"
+	"strings"
+	"sync"
+	"syscall"
+
+	"github.com/sirupsen/logrus"
+)
+
+const (
+	// KILOBYTE represents kilo bytes.
+	KILOBYTE = 1024
+
+	// MEGABYTE represents mega bytes.
+	MEGABYTE = 1024 * KILOBYTE
+
+	//GIGABYTE represents giga bytes.
+	GIGABYTE = 1024 * MEGABYTE
+
+	// TERABYTE represents tera bytes.
+	TERABYTE = 1024 * GIGABYTE
+
+	// QuotaMinID represents the minimize quota id.
+	QuotaMinID = uint32(16777216)
+)
+
+var (
+	lock        sync.Mutex
+	quotaLastID uint32
+	// UseQuota represents use quota or not.
+	UseQuota     = true
+	quotaIDs     = make(map[uint32]uint32)
+	mountPoints  = make(map[uint64]string)
+	bytesPattern = regexp.MustCompile(`(?i)^(-?\d+)([KMGT]B?|B)$`)
+	gquota       = NewQuota()
+)
+
+// BaseQuota defines the quota operation interface.
+type BaseQuota interface {
+	QuotaDriverStart(dir string) (string, error)
```

Comment:
From my own point of view, StartQuotaDriver is better than QuotaDriverStart. WDYT?

Reply:
ok